src/stream/src/executor/aggregation/state_cache/array_agg.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::array::ListValue;
+use risingwave_common::types::{Datum, DatumRef, ScalarRefImpl};
+use smallvec::SmallVec;
+
+use super::StateCacheAggregator;
+
+pub struct ArrayAgg;
+
+impl StateCacheAggregator for ArrayAgg {
+    type Value = Datum;
+
+    fn convert_cache_value(&self, value: SmallVec<[DatumRef<'_>; 2]>) -> Self::Value {
+        value[0].map(ScalarRefImpl::into_scalar_impl)
+    }
+
+    fn aggregate<'a>(&'a self, values: impl Iterator<Item = &'a Self::Value>) -> Datum {
+        let mut res_values = Vec::with_capacity(values.size_hint().0);
+        for value in values {
+            res_values.push(value.clone());
+        }
```

Comment:
https://doc.rust-lang.org/stable/src/alloc/vec/spec_from_iter_nested.rs.html#48-65

```suggestion
        let res_values = values.cloned().collect();
```